api4/plugin_test.go

```diff
@@ -387,80 +389,82 @@ func TestDisableOnRemove(t *testing.T) {
 			th := Setup(t).InitBasic()
 			defer th.TearDown()
 
-			th.App.UpdateConfig(func(cfg *model.Config) {
-				*cfg.PluginSettings.Enable = true
-				*cfg.PluginSettings.EnableUploads = true
-			})
-
-			// Upload
-			manifest, resp := th.SystemAdminClient.UploadPlugin(bytes.NewReader(tarData))
-			CheckNoError(t, resp)
-			require.Equal(t, "testplugin", manifest.Id)
+			th.TestForSystemAdminAndLocal(t, func(t *testing.T, client *model.Client4) {
+				th.App.UpdateConfig(func(cfg *model.Config) {
+					*cfg.PluginSettings.Enable = true
+					*cfg.PluginSettings.EnableUploads = true
+				})
 
-			// Check initial status
-			pluginsResp, resp := th.SystemAdminClient.GetPlugins()
-			CheckNoError(t, resp)
-			require.Empty(t, pluginsResp.Active)
-			require.Equal(t, pluginsResp.Inactive, []*model.PluginInfo{{
-				Manifest: *manifest,
-			}})
+				// Upload
+				manifest, resp := th.SystemAdminClient.UploadPlugin(bytes.NewReader(tarData))
```

Comment:
I think all `th.SystemAdminClient` in this test (`TestDisableOnRemove`) need to be changed to `client`

Reply:
Good catch! Fixed in https://github.com/mattermost/mattermost-server/pull/14543/commits/e1028b2555d4987574d469ccedb76614b8b088a8